src/Cornelis/Highlighting.hs

```diff
@@ -158,42 +151,38 @@ setHighlight b (sl, sc) (el, ec) hl = do
           $ show hl
       )
     ]
+  debug ("setHl", x, y, hl)
+  pure y
+
 
 highlightInterval
     :: Buffer
-    -> Interval' LineOffset
+    -> AgdaInterval
     -> HighlightGroup
     -> Neovim CornelisEnv (Maybe Extmark)
 highlightInterval b int hl = do
-  Interval (Pos sl sc) (Pos el ec) <- traverseInterval (vimifyPositionM b) int
-  let to_vim = subtract 1 . fromIntegral . getOneIndexedLineNumber
-  setHighlight b (to_vim sl, sc) (to_vim el, ec) hl
+  int' <- traverse (vimify b) int
+  setHighlight b int' hl
 
 
 parseExtmark :: Buffer -> Object -> Neovim CornelisEnv (Maybe ExtmarkStuff)
 parseExtmark b
   (ObjectArray ( (objectToInt -> Just ext)
-               : (objectToInt -> Just line)
-               : (objectToInt -> Just col)
+               : (objectToInt @Int -> Just (zeroIndexed -> start_line))
+               : (objectToInt @Int -> Just (zeroIndexed -> start_col))
                : ObjectMap details
                : _
                )) = runMaybeT $ do
-  vim_end_col <- hoistMaybe $ objectToInt =<< M.lookup (ObjectString "end_col") details
-  -- Plus one here because our lines are 1-indexed but the results of
-  -- get_extmarks is 0-indexed.
-  let start_line = LineNumber $ line + 1
-  end_line <- hoistMaybe $ fmap (LineNumber . (+1))
-            . objectToInt =<< M.lookup (ObjectString "end_row") details
+  end_col <- hoistMaybe $ fmap zeroIndexed
```

Comment:
Is there a better way to document which things are indexed by which means? Seems error prone to need to just need the right function call whenever doing anything with offsets.

Reply:
I guess this is the wrong place for this question since this thing parses directly from vim; but maybe  some sort of policy that we only ever do offset wrangling to wrap the vim api?

Reply:
Nah, it's fine as is; I'm just trying to wrap my head around how this all fits together.